src/common/users.ts

```diff
@@ -449,3 +449,36 @@ export enum LogoutReason {
   UserDeleted = 'user deleted',
   KratosSessionAlreadyAvailable = 'kratos session already available',
 }
+
+export const roadmapShSocialUrlMatch =
+  /^(?:(?:https:\/\/)?(?:www\.)?roadmap\.sh\/u\/)?(?<value>[\w-]{2,})\/?$/;
+
+export const twitterSocialUrlMatch =
+  /^(?:(?:https:\/\/)?(?:www\.)?(?:twitter|x)\.com\/)?@?(?<value>[\w-]{2,})\/?$/;
+
+export const githubSocialUrlMatch =
+  /^(?:(?:https:\/\/)?(?:www\.)?github\.com\/)?(?<value>[\w-]{2,})\/?$/;
+
+export const threadsSocialUrlMatch =
+  /^(?:(?:https:\/\/)?(?:www\.)?threads\.net\/)?@?(?<value>[\w-]{2,})\/?$/;
+
+export const codepenSocialUrlMatch =
+  /^(?:(?:https:\/\/)?(?:www\.)?codepen\.io\/)?(?<value>[\w-]{2,})\/?$/;
+
+export const redditSocialUrlMatch =
+  /^(?:(?:https:\/\/)?(?:www\.)?reddit\.com\/(?:u|user)\/)?(?<value>[\w-]{2,})\/?$/;
+
+export const stackoverflowSocialUrlMatch =
+  /^(?:https:\/\/)?(?:www\.)?stackoverflow\.com\/users\/(?<value>\d{2,}\/?[\w-]{2,}?)\/?$/;
+
+export const youtubeSocialUrlMatch =
+  /^(?:(?:https:\/\/)?(?:www\.)?youtube\.com\/)?@?(?<value>[\w-]{2,})\/?$/;
+
+export const linkedinSocialUrlMatch =
+  /^(?:(?:https:\/\/)?(?:www\.)?linkedin\.com\/in\/)?(?<value>[\w-]{2,})\/?$/;
+
+export const mastodonSocialUrlMatch =
+  /^(?<value>https:\/\/(?:[a-z0-9-]+\.)*[a-z0-9-]+\.[a-z]{2,}\/@[\w-]{2,}\/?)$/;
+
+export const socialUrlMatch =
+  /^(?<value>https:\/\/(?:[a-z0-9-]{1,50}\.){0,5}[a-z0-9-]{1,50}\.[a-z]{2,24}\/?[\w\-?&=]{0,100})$/;
```

Comment:
Since these don't really restrict to any domains, perhaps we should add a blacklist of (common) domains we don't want (i.e. porn sites etc)

Reply:
Since we allowed anything up until this point this is already much better. But I agree this could be cool as well. I'll look into it.

Reply:
Added task for now https://dailydotdev.atlassian.net/browse/AS-466